rcl_yaml_param_parser/test/test_yaml_variant.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <yaml.h>
+
+#include <string>
+#include <vector>
+
+#include "osrf_testing_tools_cpp/scope_exit.hpp"
+#include "rcl_yaml_param_parser/parser.h"
+#include "../src/impl/yaml_variant.h"
+#include "rcutils/allocator.h"
+#include "rcutils/strdup.h"
+#include "rcutils/testing/fault_injection.h"
+
+#define TEST_VARIANT_COPY(dest_variant, src_variant, field, tmp_var, allocator) \
+  do { \
+    SCOPED_TRACE("TEST_VARIANT_COPY " #field); \
+    src_variant.field = &tmp_var; \
+    EXPECT_TRUE(rcl_yaml_variant_copy(&dest_variant, &src_variant, allocator)); \
+    ASSERT_NE(nullptr, dest_variant.field); \
+    EXPECT_EQ(*src_variant.field, *dest_variant.field); \
+    rcl_yaml_variant_fini(&dest_variant, allocator); \
+    variant.field = nullptr; \
```

Comment:
@brawner should it be
```suggestion
    src_variant.field = nullptr; \
```
?